### PR TITLE
Update sample vptr

### PR DESCRIPTION
**Pull request by codeplay-kostasrim**
+ Removed magic constants
+ Refactored code to enhance readability
+ Removed c-style returns

### CODE REVIEW

File: samples/vptr.cpp [hunk 1]
```diff
@@ -29,92 +29,103 @@
...
+                                  access::target::global_buffer, float>(vptrA, cgh);
+      auto functor = [=](item<dim> index) {
+                       accA[index] = index[0] * multiplier1;
+                     };
+
+      cgh.parallel_for<init_a>(range<dim>{totalEntries}, functor);
     });
 
-    /* Similarly, this kernel will initialise the buffer pointed to by b. */
+    //Similarly, this kernel will initialise the buffer pointed to by b.
```
Comment (Ruyk) [inline samples/vptr.cpp]: missing space after comment

File: samples/vptr.cpp [hunk 2]
```diff
@@ -29,92 +29,103 @@
...
                                   access::target::host_buffer, float>(c_row);
-      for (size_t j = 0; j < M; j++) {
-        if (accC[row_offset + j] != (i * M + j) * (2 + 2014)) {
+      for (size_t j = 0; j < col; ++j) {
+        if (accC[row_offset + j] != (i * col + j) * (multiplier1 + multiplier2)) {
           std::cout << "Wrong value " << accC[row_offset + j] << " for element "
-                    << i * M + j << std::endl;
-          return -1;
+                    << i * col + j << '\n';
+          break;
```
Comment (Ruyk) [inline samples/vptr.cpp]: This needs to return an error code for ctest to check the correct execution of the sample.

Comment (codeplay-kostasrim) [reply samples/vptr.cpp]: done :)

File: samples/vptr.cpp (hunk 1 above)
Comment (codeplay-kostasrim) [reply samples/vptr.cpp]: done :)

**Comment by DuncanMcBain** [comment]
Closed as out of scope at the moment

**Pull request closed without merging**